models/ResourceService.cfc

```diff
@@ -490,4 +498,37 @@ component singleton accessors="true" {
 		return createObject( "java", "java.io.FileInputStream" ).init( rbFullPath );
 	}
 
+	/**
+	 * loads a java resource file from file
+	 * 
+	 * @resourceBundleFullPath full path to a (partial) resourceFile
+	 * 
+	 * @return struct resourcebundle
+	 */
+	private function _loadSubBundle( required string resourceBundleFullPath ) {
+		var resourceBundle = {};
+		var thisKey = "";
+		// create a file input stream with file location
+		var fis = getResourceFileInputStream( resourceBundleFullPath );
+		var fir = createObject( "java", "java.io.InputStreamReader" ).init( fis, "UTF-8" );
+		// init rb with file stream
+		var rb = createObject( "java", "java.util.PropertyResourceBundle").init( fir );
+		try{
+			//get keys
+			var keys = rb.getKeys();
+			//Loop through property keys and store the values into bundle
+			while( keys.hasMoreElements() ){
+				thisKey = keys.nextElement();
+				resourceBundle[ thisKey ] = rb.handleGetObject( thisKey );
+			}
+		}
+		catch( any e ) {
```

Comment:
I would say remove the catch and add a `finally` block, and this way you can consolidate the `fis.close()` into a single finally block

Reply:
Ok done. I thought we needed it for a rethrow but you are right. Only a finally block will be sufficient